### PR TITLE
FFM-7707 Add Prometheus Metrics

### PR DESCRIPTION
**Pull request by jcox250**
**What**

Adds prometheus metrics to the following places

- **HTTP Middleware**
  - This is to track total requests, request duration and the size of the request body for POST requests
- **Cache**
  - Tracking read/write operations and how long they take
- **SSE Events**
  - Tracking how many SSE events we receive and forward on to clients
- **SDK Usage**
  - Tracks which sdks are using the proxy 

**Why**

The above data would be useful for monitoring how the Proxy behaves

**Testing**

Setup a local prometheus to scrape my local proxy and was successfully scraping the metrics
Screenshot 2023-05-09 at 13 42 52

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,6 +16,7 @@ require (
 	github.com/hashicorp/go-retryablehttp v0.6.8
 	github.com/joho/godotenv v1.4.0
 	github.com/labstack/echo/v4 v4.9.0
+	github.com/prometheus/client_golang v1.11.0
```
Comment (conormurray95) [inline go.mod:19]: Is it possible to bump this to v1.11.1 instead? That version seems to have a high vulnerability in it. 
Screenshot 2023-05-12 at 10 30 02 . We can probably change this scan to fail the build if any highs exist now we're pretty clean

**Review by stephenmcconkey — APPROVED** [review]
Looks good! :)

**Review by conormurray95 — CHANGES_REQUESTED** [review]
Looks good, just bumping that one dep to avoid any security issues and we're golden

File: go.mod (hunk 1 above)
Comment (jcox250) [reply go.mod:19]: Yep will do

**Review by conormurray95 — APPROVED** [review]

**Pull request merged**